.travis.yml

```diff
@@ -0,0 +1,123 @@
+  - if [[ "${TOOLCHAIN}" == arm ]]; then
+     wget https://developer.arm.com/-/media/Files/downloads/gnu-a/8.3-2019.03/binrel/gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf.tar.xz &&
+     tar xf gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf.tar.xz;
+     export ARCH=arm;
+     export CROSS_COMPILE=$PWD/gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf/bin/arm-linux-gnueabihf-;
+     export DEFCONFIG="multi_v7_defconfig";
+   fi
+  - if [[ "${TOOLCHAIN}" == arm64 ]]; then
+      wget https://developer.arm.com/-/media/Files/downloads/gnu-a/8.3-2019.03/binrel/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu.tar.xz &&
+      tar xf gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu.tar.xz;
+      export ARCH=arm64;
+      export CROSS_COMPILE=$PWD/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu/bin/aarch64-linux-gnu-;
+      export DEFCONFIG="defconfig";
+    fi
+  - if [[ "${TOOLCHAIN}" == x86_64 ]]; then
+      wget https://mirrors.edge.kernel.org/pub/tools/crosstool/files/bin/x86_64/8.1.0/x86_64-gcc-8.1.0-nolibc-x86_64-linux.tar.gz &&
+      tar xf x86_64-gcc-8.1.0-nolibc-x86_64-linux.tar.gz;
+      export CROSS_COMPILE=$PWD/gcc-8.1.0-nolibc/x86_64-linux/bin/x86_64-linux-;
+      export DEFCONFIG="x86_64_defconfig";
+    fi
+  - popd
+  - export CROSS_COMPILE="ccache $CROSS_COMPILE"
+  - if [[ -v CHECK_COCCI ]]; then
+      sudo add-apt-repository --yes ppa:npalix/coccinelle;
+      sudo apt-get update;
+      sudo apt-get install coccinelle;
+      sudo apt-get build-dep coccinelle;
+    fi
+  - if [[ -v CHECK_SPARSE ]]; then
+      git clone --depth=1 -b v0.6.0 git://git.kernel.org/pub/scm/devel/sparse/sparse.git sparse;
+      pushd sparse;
+      make -j${BUILD_THREADS};
+      popd;
+      export SPARSE=$PWD/sparse/sparse;
+    fi
+  - echo Building ${TOOLCHAIN} with ${CROSS_COMPILE}
+
+script:
+  - make -j${BUILD_THREADS} mrproper
+  - make -j${BUILD_THREADS} ${DEFCONFIG}
+  - ./scripts/config -e COMPILE_TEST -d GENERIC_COMPAT_VDSO
+  - ./scripts/config -m DRM_CDNS_MHDP -m PHY_CADENCE_DP -m PHY_J721E_WIZ
+  - make olddefconfig
+  - make -j${BUILD_THREADS} ${BUILD_TARGETS} 2> >(tee build.log)
+  - if [[ -v CHECK_SPARSE ]]; then make C=2 CHECK=$SPARSE -j${BUILD_THREADS} ${BUILD_TARGETS} 2> >(tee sparse.log); fi
+  - if [[ -v CHECK_COCCI ]]; then make -s C=2 CHECK="scripts/coccicheck" -j${BUILD_THREADS} ${BUILD_TARGETS} | tee cocci.log; fi
+  - test ! -s build.log
+  - test ! -s sparse.log
+  - test ! -s cocci.log
+  - ccache -s
+  - cat build.log
+  - if [[ -v CHECK_SPARSE ]]; then cat sparse.log; fi
+  - if [[ -v CHECK_COCCI ]]; then cat cocci.log; fi
+
+matrix:
+  include:
+    - name: "Checkpatch"
+      before_script: skip
+      if: branch = ti-4.19.y
+      script:
+        - if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then
+            exit 0;
+          fi
+        - mkdir checkpatch_dir
+        - git format-patch ${TRAVIS_COMMIT_RANGE} -o checkpatch_dir
+        - ./scripts/checkpatch.pl --strict checkpatch_dir/*
+    - name: "Modules arm64"
```

Comment:
I had broken the matrix out to build the individual modules for individual reporting.  If 1 modules fails to build the other modules may never be built.  This is an issue we have with our product branch.  If one driver fails we don't get any reports for drivers that are built after the error.

So if DRM_CDNS_MHDP fails to build then the PHY_CADENCE may not be built especially with only 4 threads.

Reply:
True... It just felt wasting processing power, as 99% of the stuff is identical. This can be easily broken down into smaller sections if needed, though, but I thought to keep things simple for now.

And that shouldn't happen (tm). If mhdp has been merged, it should compile fine, so the only source of failures should be the branch under testing.

We could set -k for make. I haven't used that, but I presume it works fine for kernel too.